CHANGELOG.md

```diff
@@ -20,6 +20,7 @@ This project adheres to [Semantic Versioning], with the exception that minor rel
 
 ### Changed
 
+- 👷 Update C++ test matrices ([#1359]) ([**@denialhaag**])
```

Comment:
Could we be a little bit more specific here?
- we re-enabled macOS testing with gcc here by working around the respective compilation errors
- we also worked around compilation errors with clang on macOS through the latest workflow update
- and we updated from testing clang 19 and 20 to 20 and 21.